convert/convert.go

```diff
@@ -78,6 +78,14 @@ func (o *Options) Validate() error {
 	return nil
 }
 
+// Updater is an interface for tracking conversion progress.
+type Updater interface {
+	// Called from multiple goroutines after a byte range of length was converted.
+	// If the conversion is successfu, the total number of bytes will be the image
+	// virtual size.
+	Update(n int64)
```

Comment:
Can n be negative?

Reply:
No, but this is the type the go and the kernel use for file size (off_t). It is easier to use when we can use Extent.Length when updating.

We an add a check for n < 0 in the progress bar example implementation.